src/hotspot/cpu/riscv/vm_version_riscv.hpp

```diff
@@ -73,11 +73,28 @@ class VM_Version : public Abstract_VM_Version {
     virtual void update_flag() = 0;
   };
 
-  #define UPDATE_DEFAULT(flag)             \
+  #define UPDATE_DEFAULT(flag)           \
+  void update_flag() {                   \
+    assert(enabled(), "Must be.");       \
+    if (FLAG_IS_DEFAULT(flag)) {         \
+      FLAG_SET_DEFAULT(flag, true);      \
+    } else {                             \
+      /* Sync CPU features with flags */ \
+      if (!flag) {                       \
+        disable_feature();               \
+      }                                  \
+    }                                    \
+  }                                      \
+
+  #define UPDATE_DEFAULT_DEP(flag, dep)    \
   void update_flag() {                     \
       assert(enabled(), "Must be.");       \
       if (FLAG_IS_DEFAULT(flag)) {         \
```

Comment:
As this patch introduces a way to automatically disable a flag based on its dependant flag, this could introduce potential issue if the expected order is not guarantee when writing the code. It's better to have an assert there, this could be acheive by just using the existing `_value`, e.g. 
`assert(dep.value() != -1);`
Or maybe it's better to introduce a new field like `initialized`?

Reply:
That's also fine.

Reply:
Added normal assert, as it was much easier.